src/rpc/rawtransaction.cpp

```diff
@@ -316,8 +317,17 @@ static UniValue gettxchainlocks(const JSONRPCRequest& request)
         uint256 hash_block;
         int height{-1};
         bool chainLock{false};
+        bool mempool{false};
 
-        GetTransaction(nullptr, nullptr, txid, Params().GetConsensus(), hash_block);
+        const auto tx_ref = GetTransaction(nullptr, node.mempool.get(), txid, Params().GetConsensus(), hash_block);
+
+        if (tx_ref == nullptr) {
+            result.pushKV("height", height);
+            result.pushKV("chainlock", chainLock);
+            result.pushKV("mempool", mempool);
```

Comment:
no need to obscure values here imo
```suggestion
            result.pushKV("height", -1);
            result.pushKV("chainlock", false);
            result.pushKV("mempool", false);
```